### ISSUE TITLE
Devel::Declare + debugger broken since 5.13.3

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#11234]
Migrated from [rt.perl.org#87730](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=87730) (status was 'resolved')

Searchable as RT87730$

**Comment by p5pRT** [comment]
### From @schwern

#### Created by @schwern



The Method&#8203;::Signatures tests have been broken since 5\.13\.3 with a brief
fixing in 5\.13\.5\.
http&#8203;://matrix\.cpantesters\.org/?dist=Method\-Signatures

The root issue is that Devel&#8203;::Declare has been broken by 5\.13\.  It no
longer works in the debugger\.
http&#8203;://www\.cpantesters\.org/cpan/report/a68c22d2\-5abe\-11e0\-9835\-8e99bcb8094c

DD not working in the debugger means a lot of the spiffy new syntax altering
modules won't work with the debugger which makes the debugger a whole lot less
useful\.  This has been reported to DD\, but I haven't heard anything back\.
https://rt.cpan.org/Ticket/Display.html?id=67022

I suspect DD was poking at something squishy inside Perl and 5\.13 changed that\.
Either way\, I'd like Method&#8203;::Signatures to continue to be debuggable\.

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=medium

Site configuration information for perl 5.13.11:

Configured by schwern at Mon Mar 28 23:45:55 EST 2011.

Summary of my perl5 (revision 5 version 13 subversion 11) configuration:
  Commit id: 45c2973d2c99a36bfeae408ba656939295c2cb15
  Platform:
    osname=darwin, osvers=10.7.0, archname=darwin-2level
    uname='darwin windhund.local 10.7.0 darwin kernel version 10.7.0: sat jan 29 15:17:16 pst 2011; root:xnu-1504.9.37~1release_i386 i386 i386 macbookpro7,1 darwin '
    config_args='-des -Dprefix=/Users/schwern/perl5/perlbrew/perls/perl-v5.13.11-329-g45c2973 -Dusedevel'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=undef, usemultiplicity=undef
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=define, use64bitall=define, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-fno-common -DPERL_DARWIN -no-cpp-precomp -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -I/opt/local/include',
    optimize='-O3',
    cppflags='-no-cpp-precomp -fno-common -DPERL_DARWIN -no-cpp-precomp -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -I/opt/local/include'
    ccversion='', gccversion='4.2.1 (Apple Inc. build 5666) (dot 3)', gccosandvers=''
    intsize=4, longsize=8, ptrsize=8, doublesize=8, byteorder=12345678
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
    ivtype='long', ivsize=8, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='env MACOSX_DEPLOYMENT_TARGET=10.3 cc', ldflags =' -fstack-protector -L/usr/local/lib -L/opt/local/lib'
    libpth=/usr/local/lib /opt/local/lib /usr/lib
    libs=-lgdbm -ldbm -ldl -lm -lutil -lc
    perllibs=-ldl -lm -lutil -lc
    libc=/usr/lib/libc.dylib, so=dylib, useshrplib=false, libperl=libperl.a
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=bundle, d_dlsymun=undef, ccdlflags=' '
    cccdlflags=' ', lddlflags=' -bundle -undefined dynamic_lookup -L/usr/local/lib -L/opt/local/lib -fstack-protector'

Locally applied patches:
    


@INC for perl 5.13.11:
    /Users/schwern/perl5/perlbrew/perls/perl-v5.13.11-329-g45c2973/lib/site_perl/5.13.11/darwin-2level
    /Users/schwern/perl5/perlbrew/perls/perl-v5.13.11-329-g45c2973/lib/site_perl/5.13.11
    /Users/schwern/perl5/perlbrew/perls/perl-v5.13.11-329-g45c2973/lib/5.13.11/darwin-2level
    /Users/schwern/perl5/perlbrew/perls/perl-v5.13.11-329-g45c2973/lib/5.13.11
    .


Environment for perl 5.13.11:
    DYLD_LIBRARY_PATH (unset)
    HOME=/Users/schwern
    LANG=en_US.UTF-8
    LANGUAGE (unset)
    LC_CTYPE=en_US.UTF-8
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/Users/schwern/perl5/perlbrew/bin:/Users/schwern/perl5/perlbrew/perls/current/bin:/Users/schwern/bin:/Users/schwern/devel/git-achievements:/usr/local/bin:/opt/local/libexec/gnubin:/opt/local/bin:/opt/local/sbin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/usr/X11/bin
    PERL_AUTOINSTALL=--defaultdeps
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @obra




On Sun  3\.Apr'11 at 23&#8203;:29&#8203;:12 \-0700\, Michael G Schwern wrote&#8203;:
> \# New Ticket Created by  Michael G Schwern 
> \# Please include the string&#8203;:  \[perl \#87730\]
> \# in the subject line of all future correspondence about this issue\. 
> \# \<URL&#8203;: http&#8203;://rt\.perl\.org/rt3/Ticket/Display\.html?id=87730 >
> 
> The Method&#8203;::Signatures tests have been broken since 5\.13\.3 with a brief
> fixing in 5\.13\.5\.
> http&#8203;://matrix\.cpantesters\.org/?dist=Method\-Signatures
> 
> The root issue is that Devel&#8203;::Declare has been broken by 5\.13\.  It no
> longer works in the debugger\.
> http&#8203;://www\.cpantesters\.org/cpan/report/a68c22d2\-5abe\-11e0\-9835\-8e99bcb8094c
> 
> DD not working in the debugger means a lot of the spiffy new syntax altering
> modules won't work with the debugger which makes the debugger a whole lot less
> useful\.  This has been reported to DD\, but I haven't heard anything back\.
> https://rt.cpan.org/Ticket/Display.html?id=67022
> 

How small a testcase can you give us?

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @rafl

Jesse Vincent \<jesse@&#8203;fsck\.com> writes&#8203;:

> How small a testcase can you give us?

FWIW\, there's a debugger test in Devel\-Declare\. The smokers say it
passes on pretty much every perl version\.

**Comment by p5pRT** [comment]
### From @schwern

On 2011\.4\.4 9&#8203;:05 PM\, Florian Ragwitz wrote&#8203;:
> Jesse Vincent \<jesse@&#8203;fsck\.com> writes&#8203;:
> 
>> How small a testcase can you give us?
> 
> FWIW\, there's a debugger test in Devel\-Declare\. The smokers say it
> passes on pretty much every perl version\.

I've knocked it down to something just using
Devel&#8203;::Declare&#8203;::MethodInstaller&#8203;::Simple and reported that to Devel&#8203;::Declare\.
https://rt.cpan.org/Ticket/Display.html?id=67022#txn-919278

It is definitely related to how much code you inject\.  It's also extremely
finicky\.  Adding a comment at the right place makes it work\.


\-\- 
package Outer&#8203;::Space;  use Test&#8203;::More tests => 9;

**Comment by p5pRT** [comment]
### From @cpansprout

This appears to have been fixed in Devel&#8203;::Declare version 0\.006002\.

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'open' to 'resolved'
